modules/sheet/tags.js

```diff
@@ -0,0 +1,217 @@
+var {h, when, map, computed, Value, lookup} = require('mutant')
+var nest = require('depnest')
+var catchLinks = require('../../lib/catch-links')
+
+exports.needs = nest({
+  'sheet.display': 'first',
+  'keys.sync.id': 'first',
+  'contact.obs.following': 'first',
+  'contact.html.followToggle': 'first',
+  'profile.obs.rank': 'first',
+  'about.html.image': 'first',
+  'about.obs.name': 'first',
+  'app.navigate': 'first',
+  'intl.sync.i18n': 'first',
+  'tag.obs.messageTaggers': 'first'
+})
+
+exports.gives = nest('sheet.tags')
+
+exports.create = function (api) {
+  const i18n = api.intl.sync.i18n
+  const displayTags = Value(true)
+  const selectedTag = Value()
+  return nest('sheet.tags', function (msgId, ids) {
+    api.sheet.display(close => {
+      const content = computed([displayTags, selectedTag], (displayTags, selectedTag) => {
+        if (displayTags) {
+          return renderTags(ids)
+        } else {
+          return renderTaggers(msgId, selectedTag)
+        }
+      })
+      const back = computed(displayTags, (display) => {
+        if (display) {
+          return
+        } else {
+          return h('button -close', {
+            'ev-click': () => {
+              displayTags.set(true)
+              selectedTag.set()
+            }
+          }, i18n('Back'))
+        }
+      })
+      return {
+        content,
+        footer: [
+          back,
+          h('button -close', {
+            'ev-click': () => {
+              close()
+              displayTags.set(true)
+              selectedTag.set()
+            }
+          }, i18n('Close'))
+        ]
+      }
+    })
+  })
+
+  function renderTags (ids) {
+    var currentFilter = Value()
+    var tagLookup = lookup(ids, (id) => {
+      return [id, api.about.obs.name(id)]
+    })
+    var filteredIds = computed([ids, tagLookup, currentFilter], (ids, tagLookup, filter) => {
+      if (filter) {
+        var result = []
+        for (var k in tagLookup) {
+          if ((tagLookup[k] && tagLookup[k].toLowerCase().includes(filter.toLowerCase())) || k === filter) {
+            result.push(k)
+          }
+        }
+        return result
+      } else {
+        return ids
+      }
+    })
+    var content = h('div', {
+      style: { padding: '20px' }
+    }, [
+      h('h2', {
+        style: { 'font-weight': 'normal' }
+      }, [
+        i18n('Applied Tags'),
+        h('input', {
+          type: 'search',
+          placeholder: 'filter tags',
+          'ev-input': function (ev) {
+            currentFilter.set(ev.target.value)
+          },
+          hooks: [FocusHook()],
+          style: {
+            'float': 'right',
+            'font-size': '100%'
+          }
```

Comment:
this inline styling needs to come out.
Let me know if you'd like an intro to mcss at all